sections/video.liquid

```diff
@@ -100,22 +92,39 @@
         </span>
       </button>
       <template>
-        {%- if section.settings.video_url.type == 'youtube' -%}
-          <iframe
-            src="https://www.youtube.com/embed/{{ section.settings.video_url.id }}?enablejsapi=1"
-            class="js-youtube"
-            allow="autoplay; encrypted-media"
-            allowfullscreen
-            title="{{ section.settings.description | escape }}"
-          ></iframe>
+        {%- if section.settings.video == null -%}
+          {%- liquid
+            assign loop = ''
+            if section.settings.enable_video_looping
+              assign loop = '&loop=1&playlist=' | append: video_id
+            endif
+          -%}
+          {%- if section.settings.video_url.type == 'youtube' -%}
+            <iframe
+              src="https://www.youtube.com/embed/{{ video_id }}?enablejsapi=1&autoplay=1{{ loop }}"
+              class="js-youtube"
+              allow="autoplay; encrypted-media"
+              allowfullscreen
+              title="{{ section.settings.description | escape }}"
+            ></iframe>
+          {%- else -%}
+            <iframe
+              src="https://player.vimeo.com/video/{{ video_id }}?autoplay=1{{ loop }}"
+              class="js-vimeo"
+              allow="autoplay; encrypted-media"
+              allowfullscreen
+              title="{{ section.settings.description | escape }}"
+            ></iframe>
+          {%- endif -%}
         {%- else -%}
-          <iframe
-            src="https://player.vimeo.com/video/{{ section.settings.video_url.id }}"
-            class="js-vimeo"
-            allow="autoplay; encrypted-media"
-            allowfullscreen
-            title="{{ section.settings.description | escape }}"
-          ></iframe>
+          {{ section.settings.video
+            | video_tag:
+              image_size: "2048x",
```

Comment:
Hmm.. honestly I was hoping that wouldn't have to be loaded.. I see now that it is, even if I remove image_size and pass `poster: ""`. 

It renders this dom but that fallback image still loads even when not displayable..

```
<video>
  <source>
  <img /> <!-- fallback content when no html video support -->
</video>
```

So I guess the 2 options I see 
A) Keep our current responsive cover image implementation and pass a very small, inconsequential image_size to the video_tag for now.
B) Refactor the hosted video cover image to use the same fixed size that the video_tag and poster attribute will.

It's unfortunate there doesn't appear to be a way to support dynamic image sizes, but option B may be the way to go as it might have the benefit of preventing the flashing you mentioned in another comment.

Reply:
cc @ludoboludo ^

Reply:
Ended up just reducing the image size passed to the video_tag (value is debatable). I'm ok with this (for now) because that image won't impact initial load with deferred rendering. That said, I personally don't have a problem with not using the video_tag which would allow us to better control and repurpose the images used. So if anyone else agrees with that I'm happy to make that change instead.